ddtrace/appsec/_iast/_taint_tracking/Aspects/AspectExtend.cpp

```diff
@@ -19,17 +19,19 @@ api_extend_aspect(PyObject* self, PyObject* const* args, Py_ssize_t nargs)
     if (!PyByteArray_Check(to_add) and !PyBytes_Check(to_add)) {
         return nullptr;
     }
+
+    auto ctx_map = initializer->get_tainting_map();
+    const auto& to_candidate = get_tainted_object(candidate_text, ctx_map);
+    auto to_result = initializer->allocate_tainted_object(to_candidate);
```

Comment:
In order to avoid messing up to_candidate, shouldn't we use
```suggestion
    auto to_result = initializer->allocate_tainted_object_copy(to_candidate);
```